app/components/badge-github-actions.js

```diff
@@ -0,0 +1,56 @@
+import Component from '@ember/component';
+import { computed } from '@ember/object';
+import { alias } from '@ember/object/computed';
+
+export default Component.extend({
+  tagName: 'span',
+  repository: alias('badge.attributes.repository'),
+  imageUrl: computed('badge.attributes.{repository,workflow_enc,branch,event}', function() {
+    const url = new URL(`https://github.com/${this.repository}/workflows/${this.workflow_enc}/badge.svg`);
+
+    if (this.branch !== '') {
+      url.searchParams.set('branch', this.branch);
+    }
+
+    if (this.event !== '') {
+      url.searchParams.set('event', this.event);
+    }
+
+    return url.href;
+  }),
+  url: computed('badge.attributes.{repository,workflow_enc,branch,event}', function() {
+    const url = new URL(`https://github.com/${this.repository}/actions`);
+
+    if (this.workflow_enc !== '') {
+      url.searchParams.set('workflow', this.workflow_enc);
```

Comment:
It looks like this’ll double-encode the workflow.

```suggestion
      url.searchParams.set('workflow', this.workflow);
```

(and the condition too for neatness even though it behaves the same)

Reply:
According to [this doc](https://developer.mozilla.org/en-US/docs/Web/API/URL) `url.searchParams` is `readOnly`. Are you sure, that it can be used like this?

Reply:
The property is read-only:

```
> new URL('http://example.com/').searchParams = new URLSearchParams()
Uncaught:
TypeError: Cannot set property searchParams of [object URL] which has only a getter
```

but its value, the `URLSearchParams`, isn’t.

Reply:
I just tested this and the format produced by the `searchParams` isn't accepted by GitHub. We need this format: `query=workflow%3A"Say+hello%2Fwhatever"+branch%3Abadges-test+event%3Apush`